lib/providers/app_state.dart

```diff
@@ -2,94 +2,99 @@ import 'dart:async';
 import 'dart:convert';
 import 'dart:io';
 
+import 'package:flutter/foundation.dart';
 import 'package:hive/hive.dart';
-import 'package:http/http.dart' as http;
+import 'package:http/http.dart';
 import 'package:marquis_v2/env.dart';
 import 'package:marquis_v2/models/app_state.dart';
 import 'package:marquis_v2/providers/user.dart';
 import 'package:riverpod_annotation/riverpod_annotation.dart';
 
 part "app_state.g.dart";
 
-final baseUrl = environment['build'] == 'DEBUG'
-    ? environment['apiUrlDebug']
-    : environment['apiUrl'];
+final baseUrl = environment['build'] == 'DEBUG' ? environment['apiUrlDebug'] : environment['apiUrl'];
 
 @Riverpod(keepAlive: true)
 class AppState extends _$AppState {
-  Box<AppStateData>? _hiveBox;
+  final Box<AppStateData> _hiveBox;
   Timer? _refreshTokenTimer;
   Timer? _logoutTimer;
+  Client? _httpClient;
+
+  AppState({Client? httpClient, Box<AppStateData>? hiveBox})
+      : _httpClient = httpClient,
+        _hiveBox = hiveBox ?? Hive.box<AppStateData>("appState");
 
   @override
   AppStateData build() {
-    _hiveBox ??= Hive.box<AppStateData>("appState");
-    return _hiveBox!.get("appState", defaultValue: AppStateData())!;
+    _httpClient ??= Client();
```

Comment:
Set the autoLoginResult to null initially
```
return _hiveBox
        .get("appState", defaultValue: AppStateData())!
        .copyWith(autoLoginResult: null);
```

Reply:
So that the app will attempt to refresh token on start.

Current behaviour is that the token is not refreshed on init. So token errors might occur.